addon/components/frost-popover.js

```diff
@@ -1,65 +1,96 @@
 import Ember from 'ember'
+import layout from '../templates/components/frost-popover'
 import $ from 'jquery'
-import _ from 'lodash'
-import layout from './template'
-
 export default Ember.Component.extend({
-  attributeBindings: ['style'],
   layout,
-
-  event: 'manual',
-  spacing: Ember.computed.alias('offset'),
-  typeClass: 'frost-popover',
-  visibility: false,
-
-  style: Ember.computed(function () {
-    return Ember.String.htmlSafe('display:none;')
-  }),
-
-  setup: Ember.on('init', function () {
-    const parentView = this.get('parentView')
-    parentView.on('click', (event) => {
-      if (!Ember.ViewUtils.isSimpleClick(event)) {
-        return true
-      }
-
-      this.toggleProperty('visibility')
-
-      if (this.get('visibility')) {
-        Ember.run.next(this, () => {
-          $('html').on('click.container.' + this.get('elementId'), (event) => {
-            let popover = Ember.$('.tooltip-frost-popover')
-            if (!popover.is(event.target) && _.isEmpty(popover.has(event.target))) {
-              this.set('visibility', false)
-              $('html').off('click.container.' + this.get('elementId'))
-            }
-          })
+  visible: false,
+  event: 'click',
+  closest: false,
+  offset: 10,
+  position: 'bottom',
+  index: 0,
+  excludePadding: false,
+  classNameBindings: ['position', 'visible:visible:invisible'],
+  classNames: ['tooltip-frost-popover'],
+  didRender () {
+    Ember.run.next(() => {
+      const context = this
+      if (this.get('closest')) {
+        this.$().closest(this.get('target')).on(this.get('event'), function () {
```

Comment:
@sophypal Only calling `this.get('target')` and `this.get('event')` since one use is in the `if` and the other in the `else`. So only gets called once